From f8d34497dc599a338da1c9569070f7ba9c51722b Mon Sep 17 00:00:00 2001
From: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date: Fri, 21 Jul 2017 13:35:08 +0300
Subject: [PATCH] ADMINUI-2382 On-Prem- AdminUI possible bug - adding notes to
 IP's and it automatically makes duplicate entries to other ip's

---
 www/js/components/notes.jsx | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/www/js/components/notes.jsx b/www/js/components/notes.jsx
index 6f518841..1d5f8471 100644
--- a/www/js/components/notes.jsx
+++ b/www/js/components/notes.jsx
@@ -122,11 +122,21 @@ var NotesPanel = React.createClass({
 
 var NotesDropdown = React.createClass({
     getInitialState: function () {
-        return {notes: []};
+        return {
+            notes: [],
+            item: this.props.item
+        };
     },
     propTypes: {
         item: React.PropTypes.string.isRequired
     },
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.item !== this.props.item) {
+            this.setState({item: nextProps.item}, function () {
+                this.fetchNotes();
+            });
+        }
+    },
     handleUnarchive: function (item) {
         var note = new Note(item);
         var req = note.save({archived: false});
@@ -143,7 +153,7 @@ var NotesDropdown = React.createClass({
     },
     handleSave: function (item) {
         var note = new Note({
-            item_uuid: this.props.item,
+            item_uuid: this.state.item,
             note: item.note
         });
         var self = this;
@@ -205,7 +215,7 @@ var NotesDropdown = React.createClass({
         }
         var self = this;
         var collection = new Notes();
-        collection.item_uuid = this.props.item;
+        collection.item_uuid = this.state.item;
         var req = collection.fetch();
         req.done(function (notes) {
             self.setState({notes: notes});
@@ -222,7 +232,7 @@ var NotesDropdown = React.createClass({
         this.setState({dropdown: dropdown});
     },
     render: function () {
-        if (! this.props.item) {
+        if (!this.props.item) {
             console.warn('[Notes] Missing `item` property for notes component, rendering nothing');
             return null;
         }
-- 
2.21.0

